crates/pypi-types/src/requirement.rs

```diff
@@ -765,6 +761,15 @@ impl TryFrom<RequirementSourceWire> for RequirementSource {
     }
 }
 
+/// Remove the credentials from a Git URL, allowing the generic `git` username.
+pub fn redact_git_credentials(url: &mut Url) {
+    if url.username() == "git" && url.password().is_none() {
+        return;
+    }
+    let _ = url.set_password(None);
+    let _ = url.set_username("");
```

Comment:
This seems weird, shouldn't we always drop the password and conditionally drop the username? Or is this just for SSH credentials? A comment would be helpful if so.

Reply:
The password is required to be `None` in the return case though.

Reply:
Yes.. I expect:

```
let _ = url.set_password(None);
if url.username() != "git" {
    let _ = url.set_username("");
}
```

or for the documentation to properly declare that we only allow `git` if it doesn't have an associated password. Otherwise.. `git:foo@github.com` would redact both `git` and `foo`.

Reply:
Added a command + `ssh` guard.